### PR TITLE
Add update alert status to cost-management 

### PR DESCRIPTION
**Pull request by yaborens**
…anagement/preview/2018-08-01-preview/costmanagement.json

Add update alert status

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 1]
```diff
@@ -1947,6 +1947,234 @@
...
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AlertListResult"
```
Comment (anuchandy) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Can you provide an example (captured from the service) in the comment that shows how the payload for this PUT will look like?

**Review by anuchandy — CHANGES_REQUESTED** [review]

**Comment by anuchandy** [comment]
@KrisBash, @ravbhatnagar Please review from ARM side.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 2]
```diff
@@ -3091,6 +3325,46 @@
...
+          "enum": [
+			"Success",
+			"Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ModifyAlertStatus",
+            "modelAsString": true
+          },
+		  "readOnly": true
+        }
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: done

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 3]
```diff
@@ -1947,6 +1947,234 @@
...
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+	  "/providers/Microsoft.Billing/enrollmentAccounts/{enrollmentAccountId}/providers/Microsoft.CostManagement/alerts/updateStatus": {
+      "put": {
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: done

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 4]
```diff
@@ -3091,6 +3325,46 @@
...
+      "description": "The modify Alert model definition",
+      "properties": {
+        "value": {
+          "description": "Modify alert.",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Alert"
+          }
+        },        
+		"ModifyAlertStatus": {
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: done

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 5]
```diff
@@ -1947,6 +1947,234 @@
...
+            },
+            "description": "Parameters supplied to the update alerts status operation."
+          } 
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ModifyAlertListResult"
+            }
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: i will recheck this

[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: it should be: 
      "properties": {
         "Alert": {
			"description": "Modify alert.",
			"readOnly": true,
			"$ref": "#/definitions/Alert"
        }

[the remark this answers is not in the document]
Comment (anuchandy) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: @yaborens the spec says response is an array, could you share an example for request payload and it's response payload?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 6]
```diff
@@ -1947,6 +1947,234 @@
...
+    "/subscriptions/{subscriptionId}/providers/Microsoft.CostManagement/alerts/updateStatus": {
+      "put": {
+        "tags": [
+          "Alerts"
+        ],
+        "operationId": "Alerts_UpdateStatusList",
+        "description": "Update alerts status for a subscription, limit to 100 alerts.",
+        "x-ms-examples": {
+          "Alert_List": {
+            "$ref": "./examples/AlertList.json"
```
[the remark this answers is not in the document]
Comment (ravbhatnagar) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: @yaborens This API does not seem to be designed correctly. Looks like you want to allow update of some property across a collection of alerts. Is that correct? This is currently not supported. Please feel free to set up a 30 min sync with armapireview and we can explore the options.

[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: @ravbhatnagar  Yes, we would like to send a list of alerts with statuses to update.
Why this should be a problem?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: sure, working on it

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 7]
```diff
@@ -1947,6 +1947,234 @@
...
+	"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CostManagement/alerts/updateStatus": {
+      "put": {
+        "tags": [
+          "Alerts"
+        ],
+        "operationId": "Alerts_UpdateStatusByResourceGroupName",
+        "description": "Update alerts status for a resource group under a subscription. limit to 100 alerts",
+        "x-ms-examples": {
+          "Alert_List": {
+            "$ref": "./examples/AlertList.json"
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: sure, working on it

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 8]
```diff
@@ -1947,6 +1947,234 @@
...
+	"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/alerts/updateStatus": {
+      "put": {
+        "tags": [
+          "Alerts"
+        ],
+        "operationId": "Alerts_UpdateStatusByEnrollmentm",
+        "description": "Update alerts status for an enrollmentm limit to 100 alerts",
+        "x-ms-examples": {
+          "Alert_List": {
+            "$ref": "./examples/AlertList.json"
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: i will update to correct example

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 9]
```diff
@@ -1947,6 +1947,234 @@
...
+	  "/providers/Microsoft.Billing/enrollmentAccounts/{enrollmentAccountId}/providers/Microsoft.CostManagement/alerts/updateStatus": {
+      "put": {
+        "tags": [
+          "Alerts"
+        ],
+        "operationId": "Alerts_UpdateStatusListEnrollment",
+        "description": "Update alerts status for an enrollment, limit to 100 alerts",
+        "x-ms-examples": {
+          "Alert_List": {
+            "$ref": "./examples/AlertList.json"
```
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: i will update to a correct example

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 1 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: sure, I'm working on it

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: After a discussion with ravbhatnagar , the PR is on holds, we will do some changes

**Comment by anuchandy** [comment]
@yaborens if the rework that ARM suggested takes couple of days to implement in server side I would suggest to close this PR and open later once it is finalized. We have a policy, that requires closing the PR if there is no momentum.

**Pull request closed without merging**